### PR TITLE
add util to get onion address from private key

### PR DESCRIPTION
**Pull request by dballard**
Another helpful util function.

### CODE REVIEW

File: utils/crypto.go [hunk 1]
```diff
@@ -69,3 +70,16 @@ func PrivateKeyToString(privateKey *rsa.PrivateKey) string {
...
+
+// return an onion address from a private key
+func GetOnionAddress(privateKey string) (string, error) {
+	pk, _ := ParsePrivateKey([]byte(privateKey))
+	pubKey := rsa.PublicKey(pk.PublicKey)
+	addr, err := pkcs1.OnionAddr(&pubKey)
+	if err != nil {
+		return "", err
+	} else if addr == "" {
+		return "", errors.New("Could not get onion address for private key")
```
Comment (s-rah) [inline utils/crypto.go]: Add to errors.go instead of using errors.New here

File: utils/crypto.go [hunk 2]
```diff
@@ -69,3 +70,16 @@ func PrivateKeyToString(privateKey *rsa.PrivateKey) string {
 
 	return string(pem.EncodeToMemory(&privateKeyBlock))
 }
+
+// return an onion address from a private key
+func GetOnionAddress(privateKey string) (string, error) {
```
Comment (s-rah) [inline utils/crypto.go]: should have a test associated with it

Comment (s-rah) [reply utils/crypto.go]: Also why is privateKey a string?? should be rsa.PrivateKey

File: utils/crypto.go [hunk 3]
```diff
@@ -69,3 +70,15 @@ func PrivateKeyToString(privateKey *rsa.PrivateKey) string {
 
 	return string(pem.EncodeToMemory(&privateKeyBlock))
 }
+
+// return an onion address from a private key
+func GetOnionAddress(privateKey *rsa.PrivateKey) (string, error) {
+	pubKey := rsa.PublicKey(privateKey.PublicKey)
```
Comment (s-rah) [inline utils/crypto.go]: is this cast necessary?

**Comment by dballard** [comment]
derp no. fixed

**Pull request merged**